src/test/java/teammates/test/driver/HtmlHelper.java

```diff
@@ -22,53 +22,31 @@
 public class HtmlHelper {
 
     /**
-     * Verifies that two HTML files are logically 
-     * equivalent e.g., ignores differences in whitespace and attribute order.
+     * Verifies that two HTML files are logically equivalent, e.g. ignores
+     * differences in whitespace and attribute order.
+     * @param expectedString the expected string for comparison
+     * @param actualString the actual string for comparison
+     * @param isPart if true, ignores top-level HTML tags, i.e <code>&lt;html&gt;</code>,
+     *               <code>&lt;head&gt;</code>, and <code>&lt;body&gt;</code>
+     * @param isDifferenceToBeShown if true, whenever the assertion fails,
+     *                              <code>AssertionError</code> will be thrown and
+     *                              the difference can then be traced
      */
-    //TODO: for the following 4 methods, change the order of parameters passed in
-    //      should be expectedString, acutalString
-    public static void assertSameHtml(String actualString, String expectedString){       
-        String processedExpectedHtml = convertToStandardHtml(expectedString, false);
-        String processedActualHtml = convertToStandardHtml(actualString, false);
-        
-        if(!AssertHelper.isContainsRegex(processedExpectedHtml, processedActualHtml)){
-            processedActualHtml = AppPage.processPageSourceForFailureCase(processedActualHtml);
-            processedExpectedHtml = AppPage.processPageSourceForFailureCase(processedExpectedHtml);
-            assertEquals("<expected>\n"+processedExpectedHtml+"</expected>", "<actual>\n"+processedActualHtml+"</actual>");
-        }
-    }
-    
-    public static void assertSameHtmlPart(String actualString, String expectedString) {
-        String processedExpectedHtmlPart = convertToStandardHtml(expectedString, true);
-        String processedActualHtmlPart = convertToStandardHtml(actualString, true);
-        
-        if(!AssertHelper.isContainsRegex(processedExpectedHtmlPart, processedActualHtmlPart)){
-            processedActualHtmlPart = AppPage.processPageSourceForFailureCase(processedActualHtmlPart);
-            processedExpectedHtmlPart = AppPage.processPageSourceForFailureCase(processedExpectedHtmlPart);
-            assertEquals("<expected>\n"+processedExpectedHtmlPart+"</expected>", "<actual>\n"+processedActualHtmlPart+"</actual>");
-        }
-    }
+    public static boolean assertSameHtml(String expected, String actual, boolean isPart,
+                                         boolean isDifferenceToBeShown) {
+        String processedExpected = convertToStandardHtml(expected, isPart);
+        String processedActual = convertToStandardHtml(actual, isPart);
```

Comment:
Consider making this private and having separate public method with descriptive names rather than boolean flags, if the number of methods are not too high. At least get rid of one flag. However, those public methods can call the same private method with the correct boolean parameter.
The flags are an internal optimization the clients of the class need not bother with.